app/Services/Files/FileStorageService.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Services\Files;
+
+use Illuminate\Http\UploadedFile;
+use Storage;
+use Str;
+
+class FileStorageService
+{
+    public function storePubliclyWithRandomFilename(string $path, UploadedFile $file): string
+    {
+        $filename = $this->generateRandomFilename($path);
+
+        $storedPath = Storage::disk('public')->putFileAs($path, $file, "{$filename}.{$file->extension()}");
+
+        return "storage/{$storedPath}";
+    }
+
+    private function generateRandomFilename(string $path): string
```

Comment:
This method is specific to the public filesystem with its collision check, so should either:
- Be renamed for specificity the public filesystem (e.g. `generateRandomFilenameForPublicFilesystem`)
- Take a filesystem as its argument to check for a collision there, making it non-filesystem specific